### PR TITLE
Added README detailing how to configure maestro

### PR DESCRIPTION
**Pull request by mray190**
Not complete! Takes care of:
* DeviceDB connection
* SysLog injection
* Logging
* Network interfaces

### CODE REVIEW

File: configs/README.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
+# Maestro Config
+
+The following guide documents the options available in the `maestro.config` yaml file that is provided to maestro on startup.
```
Comment (costanic) [inline configs/README.md]: how is the file provided to maestro on startup?  a command line option?  hard-coded filename?

File: configs/README.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
```
Comment (costanic) [inline configs/README.md]: Maybe document the expected structure first, perhaps providing an example minimal config.

> `---`
> ` config_end: true`

File: configs/README.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+### SysLog
+
+To enable syslog, add to your `maestro.config`:
+
+```
+sysLogSocket: /run/systemd/journal/syslog
+```
+
+Where `/run/systemd/journal/syslog` is the path to your syslog file.
```
Comment (costanic) [inline configs/README.md]: syslog socket, not file.

File: configs/README.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+To inject a log into a log target, you can use syslog:
+
+```
+echo "test err message" | systemd-cat -p err
+```
+
+Where the contents of the `echo` is the message, and `err` is the log level. Note that you must have the log level you specify match what is in your log target filters list otherwise it will not show up.
+
+### Targets
+
```
Comment (costanic) [inline configs/README.md]: briefly describe what is a target

File: configs/README.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+Where `/run/systemd/journal/syslog` is the path to your syslog file.
+
+To inject a log into a log target, you can use syslog:
+
+```
+echo "test err message" | systemd-cat -p err
+```
+
+Where the contents of the `echo` is the message, and `err` is the log level. Note that you must have the log level you specify match what is in your log target filters list otherwise it will not show up.
```
Comment (CristianPrundeanuARM) [inline configs/README.md]: It would be useful to have a list of valid log levels here as well.

File: configs/README.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+```
+targets:
+    - <target 1> # Outlined below
+    - <target 2> # Outlined below
+    - <target ...> # Outlined below
+```
+
+Breakdown:
+* `name` - Unique identifier for the log target. `toCloud` is a special name for sending data to the cloud
```
Comment (CristianPrundeanuARM) [inline configs/README.md]: One thing that confused me is that the name is not required for file targets. We could clarify that here.

File: configs/README.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+* `file` - Required for log targets destined to a file. This is the output location for said file.
+* `format_time` - REQUIRED. Used to specify the time format in the output logs
+* `delim` - OPTIONAL. Used to specify the deliminator between logs
```
Comment (CristianPrundeanuARM) [inline configs/README.md]: Delimiter.
(Deliminator sounds like cool industrial machinery though...)

File: configs/README.md [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+* `filters` - REQUIRED. Used to specify what level of logs make it to the output log
+    * `warn`
+    * `info` or `success`. They behave the same
+    * `error`
+    * `debug`
+    * `all`
+* `rotate` - Optional for log targets destined to a file. Determines how to rotate between X amount of files
```
Comment (CristianPrundeanuARM) [inline configs/README.md]: I'd reword this, the current wording makes it sound like it's optional when logging to a file, but mandatory otherwise.
Maybe this: "Optional; only used for log targets destined to a file."

File: docs/maestro_config.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+    - <target 1> # Outlined below
+    - <target 2> # Outlined below
+    - <target ...> # Outlined below
+```
+
+Breakdown:
+* `name` - Unique identifier for the log target. `toCloud` is a special name for sending data to the cloud and is the required `name` for cloud targets. File targets do not need a `name`
+* `file` - Required for log targets destined to a file. This is the output location for said file.
+* `format_time` - REQUIRED. Used to specify the time format in the output logs
+* `delim` - OPTIONAL. Used to specify the delimeter between logs
```
Comment (CristianPrundeanuARM) [inline docs/maestro_config.md]: Delimiter.
